### ISSUE TITLE
[Bug] When the sampling rate is 0 and the code exists across goroutine, ExitSpan has a problem of failure to create. err: span type is wrong

### ISSUE THREAD

**Issue by Donghui0** [apache/skywalking#12079]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Go Agent (apache/skywalking-go)

### What happened

In the following scenario, Gin handler serves as the entry point and calls 2 mongo operations through goroutine.

Gin --  EntrySpan  (NewNoopSpan() -> SET_GLS(NoopSpan{stackCount: 1}))
    | 
(sync.Group)
MongoClient  ExitSpan  (GET_GLS() ->  NoopSpan{stackCount:0} -> noop.enterNoSpan() -> span.End() -> SET_GLS(nil) )
    |                                                                 (Why stackCount is 0 ?)
MongoClient  ExitSpan  (GET_GLS() -> nil)

In the 2nd ExitSpan, ctx.ActiveSpan() returns nil, Then createSpan0() returns NoopSpan.
![image](https://github.com/apache/skywalking/assets/7134928/eb5fdd24-b5f2-447c-bbde-bdcfce19e888)

The following is debug mode:
**Gin CreateEntrySpan，ctx:**
![image](https://github.com/apache/skywalking/assets/7134928/7e9ad641-2783-4bad-91b1-0f5d14c5a693)

**The first CreateExitSpan within the goroutine, ctx:**
![image](https://github.com/apache/skywalking/assets/7134928/4dde005b-331b-45ec-8ebd-f8ccd5ccc0c2)

**Second CreateExitSpan inside goroutine, ctx:**
![image](https://github.com/apache/skywalking/assets/7134928/2103ac11-78f6-4a6b-ae64-2a3069bb1d9e)

### What you expected to happen

In the first ExitSpan, The stackCount of ActiveSpan() obtained by GetTracingContext() should be 1.
in the 2nd  ExitSpan, getTracingContext() -> ActiveSpan() Should not be nil.

### How to reproduce

After changing the sampling rate to 0, run the following test code.
```
package main

import (
    "fmt"
    _ "github.com/apache/skywalking-go"
    "github.com/gin-gonic/gin"
    "go.mongodb.org/mongo-driver/bson"
    "go.mongodb.org/mongo-driver/mongo"
    "go.mongodb.org/mongo-driver/mongo/options"
    "golang.org/x/sync/errgroup"
)

type ErrorGroup struct {
    errgroup.Group
}

func main() {
    r := gin.Default()
    r.GET("/mongo_get", mongoGet)
    r.Run()
}

func mongoGet(c *gin.Context) {
    var e ErrorGroup
    var m, _ = mongo.Connect(c, options.Client().ApplyURI("mongodb://127.0.0.1:27017"))
    filter := bson.D{{"name", "test"}}
    e.Go(func() (err error) {
       collection := m.Database("demo").Collection("testCollection")
       _, err = collection.Find(c, filter)
       if err != nil {
          return err
       }
       fmt.Println("first Find Done")
       _, err = collection.Find(c, filter)
       if err != nil {
          return err
       }
       fmt.Println("second Find Done")
       return nil
    })
    err := e.Wait()
    if err != nil {
       fmt.Println(err)
    }
}
```

Observe the log, you can find that the log will print the following error:
`cannot create exit span on mongo client: span type is wrong`

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Which version are you using? There was a relative fix in 0.4 release, https://github.com/apache/skywalking-go/pull/161

**Comment by Qiangel** [comment]
> > Which version are you using? There was a relative fix in 0.4 release, [apache/skywalking-go#161](https://github.com/apache/skywalking-go/pull/161)
> 
> version 0.4.0

Please check gmail to discuss the same issue，ty

**Comment by wu-sheng** [comment]
Recheck 0.5. It should have been fixed.
